core/src/main/java/org/apache/druid/storage/StorageConnector.java

```diff
@@ -104,4 +105,6 @@
    * @throws IOException
    */
   void deleteRecursively(String path) throws IOException;
+
+  List<String> ls(String dirName) throws IOException;
```

Comment:
Can we make the method name more readable, `listFiles` maybe? Javadocs and unit tests would be useful for this method as well.